superset-frontend/src/cccs-viz/plugins/plugin-chart-ag-grid/src/ag-grid/AGGridViz.tsx

```diff
@@ -355,39 +354,45 @@ export default function AGGridViz({
     let specialMenuItems: JSX.Element[] = [];
     if (
       enableAlfred &&
-      selectedData.advancedTypeData.harmonized_email_id &&
-      selectedData.advancedTypeData.harmonized_email_id.length > 0
+      selectedData.typeData.harmonized_email_id &&
+      selectedData.typeData.harmonized_email_id.length > 0
     ) {
       specialMenuItems = [
         ...specialMenuItems,
         <RetainEmlMenuItem
           onSelection={handleContextMenu}
           label="Retain EML Record To ALFRED"
           key="retain-eml"
-          data={selectedData.advancedTypeData.harmonized_email_id}
-          disabled={
-            selectedData.advancedTypeData.harmonized_email_id.length > 30
-          }
+          data={{
+            email_ids: selectedData.typeData.harmonized_email_id,
+            dates: [
+              ...(selectedData.typeData['TIMESTAMP WITH TIME ZONE'] || []),
+              ...(selectedData.typeData['TIMESTAMP WITHOUT TIME ZONE'] || []),
+              ...(selectedData.typeData.DATE || []),
+              ...(selectedData.typeData.DATETIME || []),
+            ],
+          }}
+          disabled={selectedData.typeData.harmonized_email_id.length > 100}
```

Comment:
@cccs-rc mentioned it, that limit of 100 should be a constant, the same constant could be reused 3 times here and avoid issues in the future.

Reply:
my bad @cccs-rc i missed the end of your comment. Good idea I changed it to a constant in this file